components/application-role-mgt/org.wso2.carbon.identity.application.role.mgt/src/main/java/org/wso2/carbon/identity/application/role/mgt/dao/impl/ApplicationRoleMgtDAOImpl.java

```diff
@@ -0,0 +1,475 @@
+            }
+            ApplicationRole applicationRole = new ApplicationRole(roleId);
+            applicationRole.setAssignedUsers(users);
+            return  applicationRole;
+        } catch (DataAccessException e) {
+            throw handleServerException(ERROR_CODE_GET_ROLE_ASSIGNED_USERS, e, roleId);
+        }
+    }
+
+    @Override
+    public void updateApplicationRoleAssignedGroups(String roleId, IdentityProvider identityProvider,
+                                                    List<String> addedGroups, List<String> removedGroups,
+                                                    String tenantDomain)
+            throws ApplicationRoleManagementException {
+
+        validateGroupIds(identityProvider, addedGroups, tenantDomain);
+        validateGroupIds(identityProvider, removedGroups, tenantDomain);
+        NamedJdbcTemplate namedJdbcTemplate = getNewTemplate();
+        try {
+            namedJdbcTemplate.withTransaction(template -> {
+                if (addedGroups.size() > 0) {
+                    namedJdbcTemplate.executeBatchInsert(SQLConstants.ADD_APPLICATION_ROLE_GROUP,
+                            (preparedStatement -> {
+                                for (String groupId : addedGroups) {
+                                    preparedStatement.setString(DB_SCHEMA_COLUMN_NAME_ROLE_ID, roleId);
+                                    preparedStatement.setString(DB_SCHEMA_COLUMN_NAME_GROUP_ID, groupId);
+                                    preparedStatement.setString(DB_SCHEMA_COLUMN_NAME_IDP_ID,
+                                            identityProvider.getResourceId());
+                                    preparedStatement.setInt(DB_SCHEMA_COLUMN_NAME_TENANT_ID,
+                                            getTenantId(tenantDomain));
+                                    preparedStatement.addBatch();
+                        }
+                    }), roleId);
+                }
+                if (removedGroups.size() > 0) {
+                    for (String groupId: removedGroups) {
+                        namedJdbcTemplate.executeUpdate(SQLConstants.DELETE_ASSIGNED_GROUP_APPLICATION_ROLE,
+                                namedPreparedStatement -> {
+                                    namedPreparedStatement.setString(DB_SCHEMA_COLUMN_NAME_ROLE_ID, roleId);
+                                    namedPreparedStatement.setString(DB_SCHEMA_COLUMN_NAME_GROUP_ID, groupId);
+                                    namedPreparedStatement.setInt(DB_SCHEMA_COLUMN_NAME_TENANT_ID,
+                                            getTenantId(tenantDomain));
+                                });
+                    }
+                }
+                return null;
+            });
+        } catch (TransactionException e) {
+            if (checkUniqueKeyConstrainViolated(e)) {
+                throw ApplicationRoleMgtUtils.handleClientException(ERROR_CODE_GROUP_ALREADY_ASSIGNED, roleId);
```

Comment:
Shall we add the static import for client exception handling method as well?

Reply:
fixed